lib/jnpr/junos/utils/start_shell.py

```diff
@@ -100,18 +112,20 @@ def open(self):
         """
         if self.ON_JUNOS is True:
             self._chan = subprocess.Popen(
-                ["cli", "start", "shell"],
+                ["cli", "start", "shell", self.shell_type],
                 shell=False,
                 stdin=subprocess.PIPE,
                 stdout=subprocess.PIPE,
+                close_fds=1,
```

Comment:
Where did you create fd ?

Reply:
removed the close_fds and bufsize
            self._chan = subprocess.Popen(
                ["cli", "start", "shell", self.shell_type],
                shell=False,
                stdin=subprocess.PIPE,
                stdout=subprocess.PIPE,
            )